tuf/api/metadata.py

```diff
@@ -1301,3 +1301,32 @@ def to_dict(self) -> Dict[str, Any]:
     def update(self, fileinfo: TargetFile) -> None:
         """Assigns passed target file info to meta dict."""
         self.targets[fileinfo.path] = fileinfo
+
+    def add_key(self, role: str, key: Key) -> None:
+        """Adds new signing key for delegated role 'role'.
+
+        Raises:
+            UnknownRoleError: If there are no 'delegations' in the Targets obj.
+            KeyError: If there is no role 'role' in the Delegations object.
+        """
+        if self.delegations is None:
+            raise exceptions.UnknownRoleError("Delegations doesn't exists!")
+        self.delegations.roles[role].keyids.add(key.keyid)
```

Comment:
* now we get two different errors when a nonexisting role is used (UnknownRoleError if delegations does not exist, and KeyError if it does). This is silly: no user of the API wants that.
* Also (@sechkova knows me) we should only use custom errors when they provide actual value -- when callers might want to catch specifically that error and the base library errors would be too vague

There's basically two slightly overlapping classes of exceptions:

1. the ones we expect the caller to handle: these should be carefully designed and documented (and here custom exceptions _may_ provide added value)
2. the ones that represent caller programming error: these especially should be standard base library exceptions.

this case is mostly in group 2: adding a key for a non-existing role is unlikely to be handled in calling code. The "correct" error would be ValueError I guess but the two nonexisting role cases listed above should use the same error -- in that context I don't think KeyError is a bad choice (since we get that for free when delegations does exist).

Reply:
also, the error here (from callers perspective) isn't _"Delegations doesn't exists!"_: that's an implementation detail. The error is that caller tried to add a key to a role that does not exist.

Reply:
Maybe the best solution is just to use `KeyError` as it's the simplest one and we are already throwing it as you said.